stable/pomerium/templates/proxy-deployment.yaml

```diff
@@ -53,26 +55,22 @@ spec:
         - name: COOKIE_SECRET
           valueFrom:
             secretKeyRef:
-              name: {{ if not .Values.config.existingSecret }}{{ template "pomerium.fullname" . }}{{ else }}{{ .Values.config.existingSecret }}{{ end }}
+              name: {{ $configSecretName }}
               key: cookie-secret
         - name: SHARED_SECRET
           valueFrom:
             secretKeyRef:
-              name: {{ if not .Values.config.existingSecret }}{{ template "pomerium.fullname" . }}{{ else }}{{ .Values.config.existingSecret }}{{ end }}
+              name: {{ $configSecretName }}
               key: shared-secret
         - name: AUTHENTICATE_SERVICE_URL
-          value: {{ .Values.proxy.authenticateServiceUrl }}
+          value: {{ default (printf "https://%s.%s" .Values.authenticate.name .Values.config.rootDomain ) .Values.proxy.authenticateServiceUrl }}
         - name: AUTHORIZE_SERVICE_URL
-          value: {{ .Values.proxy.authorizeServiceUrl }}
-{{- if .Values.proxy.authenticateUseInternalUrl }}
+          value: {{ default (printf "https://%s.%s" .Values.authorize.name .Values.config.rootDomain ) .Values.proxy.authorizeServiceUrl }}
         - name: AUTHENTICATE_INTERNAL_URL
-          value: {{ template "pomerium.fullname" . }}-{{ .Values.authenticate.name }}
-{{- end }}
-{{- if .Values.proxy.authorizeUseInternalUrl }}
+          value: {{ default (printf "%s-%s.%s.svc.cluster.local" (include "pomerium.fullname" .) .Values.authenticate.name .Release.Namespace ) .Values.proxy.authenticateUseInternalUrl}}
```

Comment:
@desimone is this working? If authenticateUseInternalUrl is set (to false or true), it should use its value… no?

Reply:
@desimone ah but maybe you weren't planning for `authenticateUseInternalUrl` to be a boolean. So if it's an empty string, I think `default` will use the default alternative, not the empty string.

Also, if it's a string, I think it should be named `authenticateInternalUrl` (without the `Use` word).

Reply:
Good catch.